synapse/handlers/relations.py

```diff
@@ -438,11 +390,11 @@ async def get_bundled_aggregations(
         # events to be fetched. Thus, we check those first!
 
         # Fetch thread summaries (but only for the directly requested events).
-        #
-        # Note that you can't have threads off of other related events, but it is
-        # possible for a malicious homeserver to inject them anyway.
         threads = await self.get_threads_for_events(
-            events_by_id.keys(), user_id, ignored_users
+            # It is not valid to start a thread on an event which already has a relation.
```

Comment:
Currently this somehow implies to me an event which is the *target* of a relation.

```suggestion
            # It is not valid to start a thread on an event which itself relates to another event.
```